Neos.Flow/Classes/Core/Booting/Scripts.php

```diff
@@ -673,16 +673,16 @@ public static function initializeResources(Bootstrap $bootstrap)
      *
      * @param string $commandIdentifier E.g. neos.flow:cache:flush
      * @param array $settings The Neos.Flow settings
-     * @param boolean $outputResults if false the output of this command is only echoed if the execution was not successful
+     * @param boolean $outputResults Echo the commands output on success
      * @param array $commandArguments Command arguments
-     * @return boolean true if the command execution was successful (exit code = 0)
+     * @param array|null &$output If the output argument is present, then the specified array will be filled with every line of output from the command. {@see exec()}
```

Comment:
```suggestion
     * @param array &$output If the output argument is present, then the specified array will be filled with every line of output from the command. {@see exec()}
```

Reply:
> so you can pass an uninitialized variable

_That_ is ugly as hell. I did not understand what you were saying first, as I never did that. I always initialize variables… but yes, you are right. Passing in an unitialized variable means passing `null`, and that needs to be allowed then.

I still prefer

```php
$output = [];
Scripts::executeCommand(..., $output);
```

over

```php
Scripts::executeCommand(..., $output);
```

Especially as `$output` **is appended to** by `exec()`, so explicitly initializing such variables is safer. 

But hey, do as you like. In this case, following "accepted practices" seems reasonable. 👍

Reply:
👍 actually, hate to tell you, but in flow and neos we also usually pass in `$matches` uninitialized:

https://github.com/neos/flow-development-collection/blob/24297f3ab69357855d243a52a16bc0e06f725720/Neos.Flow/Classes/ResourceManagement/ResourceManager.php#L456-L465

and im using this approach also when writing new code ^^